Branch-SDK/src/io/branch/referral/ServerResponse.java

```diff
@@ -135,4 +140,28 @@ public JSONArray getArray() {
 	public BranchLinkData getLinkData() {
 		return linkData_;
 	}
+
+	/**
+	 * <p>Sets the request object caused this response, as a generic {@link Object}
+	 * instance. This object can be type-cast by other methods within this class:</p>
+	 *
+	 * @param request Generic {@link Object} instance containing the request data caused this response.
+	 */
+	public void setRequestObject(Object request) {
+		this.requestObject_ = request;
+	}
+
+	/**
+	 * <p>Checks whether the request params cause this response is an instance of a
+	 * {@link JSONObject} object, and if so type-casts it to the corresponding class type.</p>
+	 *
+	 * @return A {@link JSONObject} containing the post params sent with corresponding  server request, or a null
+	 * value if the post params is not of the {@link JSONObject} type.
+	 */
+	public JSONObject getRequestObject() {
```

Comment:
I'm not really sure I understand what you're saying here, but based on [the way you use it here](https://github.com/BranchMetrics/Branch-Android-SDK/pull/76/files#diff-11fc8f6026de99b60216e9aca37d6d26R3173) , it looks to me like you're treating it as a response body, right?

Or, are you adding the request object onto the response, and then inspecting fields on the request? If so, I'd really prefer we avoid that. I specifically removed that same behavior from the iOS SDK already -- we want to avoid making every class dependent on every other class; there should be a better way to keep track of the amount redeemed.

If this is all you can do for this pull request, that's ok, but I really want to avoid this in the future.